src/utils/TokenUtils.ts

```diff
@@ -65,16 +65,29 @@ export const resolveContractFromSymbol = (
 };
 
 export function getTokenInformationFromAddress(address: string, tokenMapping = TOKEN_SYMBOLS_MAP): L1Token | undefined {
-  const details = Object.values(tokenMapping).find((details) => {
+  let details = Object.values(tokenMapping).find((details) => {
```

Comment:
If you change this to an Object.entries() can you skip the code block on lines 77-81?

Reply:
I should have done that. It's refactored now so `chainId` is a function input.